### PR TITLE
[Iwara] Fix download

### PR DESCRIPTION
**Pull request by hakatashi**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

[Iwara](http://www.iwara.tv/) now switched their player to get media URLs from API, not HTML. So the previous extractor is now [failing](https://travis-ci.org/rg3/youtube-dl/jobs/192039917#L892).

This patch fixes the extractor to use API to extract video URL and fixed some test cases as the following.

* Some MD5 are updated according to the source change. I manually confirmed the downloaded content is correct.
* The video of the last test case was re-rated to 18+, and I updated the field.
* Then there are no test cases for 0+ video (which has `www.iwara.tv` domain), and I added one.

### CODE REVIEW

File: youtube_dl/extractor/iwara.py [hunk 1]
```diff
@@ -52,26 +61,72 @@ def _real_extract(self, url):
...
-        info_dict.update({
+            if raw_format.get('resolution') == '1080p':
+                height = 1080
+            elif raw_format.get('resolution') == '720p':
+                height = 720
+            elif raw_format.get('resolution') == '540p':
+                height = 540
+            elif raw_format.get('resolution') == '360p':
+                height = 360
+            else:
```
Comment (dstftw) [inline youtube_dl/extractor/iwara.py]: Code duplication.

File: youtube_dl/extractor/iwara.py [hunk 2]
```diff
@@ -52,26 +61,72 @@ def _real_extract(self, url):
...
+            api_url, video_id,
+            note='Downloading formats')
+
+        formats = []
+
+        for raw_format in raw_formats:
+            new_format = {
+                'url': raw_format['uri'],
+                'resolution': raw_format['resolution'],
+                'format_id': raw_format['resolution'],
```
Comment (dstftw) [inline youtube_dl/extractor/iwara.py]: Should not break if there is no `resolution` key.

File: youtube_dl/extractor/iwara.py [hunk 3]
```diff
@@ -52,26 +61,72 @@ def _real_extract(self, url):
...
+                height = 360
+            else:
+                height = None
+
+            if height is not None:
+                new_format['width'] = int(height / 9.0 * 16.0)
+                new_format['height'] = height
+
+            if raw_format.get('mime') == 'video/mp4':
+                new_format['ext'] = 'mp4'
```
Comment (dstftw) [inline youtube_dl/extractor/iwara.py]: `utils.mimetype2ext`.

File: youtube_dl/extractor/iwara.py (hunk 1 above)
Comment (hakatashi) [inline youtube_dl/extractor/iwara.py]: Thanks. How should I fix this?

**Comment by yan12125** [comment]
@hakatashi Sorry I didn't notice here's a pull request and I've push my patch to caf0f5f8b7d0854caaf6778fe3a646ee0d7668fe and pick some codes from this PR in 2ab2c0d1f53f66614eda4fefb042e851e78097f0. Anyway thanks for the contribution!

**Pull request closed without merging**